internal/xdscache/v3/listener.go

```diff
@@ -485,11 +498,13 @@ func (c *ListenerCache) OnChange(root *dag.DAG) {
 			// Secret is provided when TLS is terminated and nil when TLS passthrough is used.
 			if vh.Secret != nil {
 				// Choose the higher of the configured or requested TLS version.
-				vers := max(cfg.minTLSVersion(), envoy_v3.ParseTLSVersion(vh.MinTLSVersion))
+				minVer := max(cfg.minTLSVersion(), envoy_v3.ParseTLSVersion(vh.MinTLSVersion))
+				maxVer := max(cfg.maxTLSVersion(), envoy_v3.ParseTLSVersion(vh.MaxTLSVersion))
```

Comment:
The logic for minimum TLS version enforces that the minimum TLS version on a specific HTTPProxy is _at least_ as high as the min TLS version configured in the global Contour config.

For max TLS version, I think the corresponding logic would be that the global Contour config defines a ceiling on the max TLS version, and individual HTTPProxies can't exceed that max -- e.g. if the global config says max TLS version = 1.2, and a specific proxy says 1.3, then the effective max TLS version for that proxy is 1.2.

So, we'd want to use a `min` function to get the effective max TLS version.

cc @sunjayBhatia -- does this sound reasonable to you, given how the min TLS version logic works?

Reply:
@sunjayBhatia  WDYT?

Reply:
yep the logic @skriss describes sounds reasonable to me 👍🏽

Reply:
ok